File: server/models/Course.js
// models/Course.js
const mongoose = require('mongoose');

const courseSchema = new mongoose.Schema({
  title: { type: String, required: true },
  subtitle: String,
  thumbnail: String,
  category: String,
  difficulty: String,
  duration: String,
  modules: Number,
  badges: [String],
  instructor: {
    name: String,
    title: String,
    avatar: String,
  },
  externalLink: String,
}, { timestamps: true });

module.exports = mongoose.model('Course', courseSchema);

File: server/models/course.py
from mongoengine import Document, StringField, IntField, ListField, EmbeddedDocument, EmbeddedDocumentField

class Instructor(EmbeddedDocument):
    name = StringField()
    title = StringField()
    avatar = StringField()

class Course(Document):
    title = StringField(required=True)
    subtitle = StringField()
    thumbnail = StringField()
    category = StringField()
    difficulty = StringField()
    duration = StringField()
    modules = IntField()
    badges = ListField(StringField())
    instructor = EmbeddedDocumentField(Instructor)
    externalLink = StringField()

    meta = {'collection': 'courses'}
